common/downstream_fatal_error.md

```diff
@@ -0,0 +1,131 @@
+---
+layout: default
+parent: Common
+---
+
+# Downstream Fatal Error
+
+Understanding the `DownstreamFatalError` exception.
+{: .fs-6 .fw-300 }
+
+&nbsp;
+
+{: .note }
+> _pyRACF expects IRRSMO00 to return a string containing XML data. If IRRSMO00 returns an empty string, or if the **SAF Return Code** in the result XML is greater than `4`, `DownstreamFatalError` is raised._
+
+&nbsp;
+
+pyRACF expects IRRSMO00 to return a non-empty response string after processing a request. If IRRSMO00 returns an empty response string or the **SAF Return Code** is greater than `4`, indicating that there was an issue with command image processing, a `DownstramFatalError` will be raised. It is possible that RACF commands are executed in this situation, but this is not the case for any known causes of this error.
+
+## RACF Authorizations
+
+&nbsp;
+
+A common cause of this error is that the user does not have the proper RACF authorizations as outlined in [our dependencies note](../../index). One of the possible failures is the user not having at least `READ` authority to the `IRR.IRRSMO00.PRECHECK` resource in the `XFACILIT` class, which is required for `set` or `alter` operations. See the [Check for & set up RACF Authorizations](../setup_precheck) documentation for more details. Another possibility is the user not having `ALTER` authority to the `userid.IRRSMO00` resource in the `SURROGAT` class, which is required to make security requests as another userid. Please review our [run as userid](../run_as_userid) documentation for more information on this feature.
+
+###### Python Script
+```python
+from pyracf import UserAdmin
+from pyracf import DownstreamFatalError
+
+user_admin = UserAdmin()
+
+try:
+    user_admin.set_uid("squidwrd", 123456)
+except DownstreamFatalError as e:
+    print(e.message)
+```
+
+###### Console Output
+```console
+Security request made to IRRSMO00 failed.
+
+SAF Return Code: 8 
+RACF Return Code: 200 
+RACF Reason Code: 16
+
+Check to see if the proper RACF permissions are in place.
+For `set` or `alter` functions, you must have at least READ access to `IRR.IRRSMO00.PRECHECK` in the `XFACILIT` class.
+```
+
+###### Python Script
+```python
+from pyracf import UserAdmin
+from pyracf import DownstreamFatalError
+
+user_admin = UserAdmin(run_as_userid="ESWIFT")
+
+try:
+    user_admin.set_uid("squidwrd", 123456)
+except DownstreamFatalError as e:
+    print(e.message)
+```
+
+###### Console Output

```

Comment:
This sample seems to be duplicated.

Reply:
the samples here are marginally different, in the first, the "shown" outcome is a lack of precheck permissions, so the user is "trying" to do a set_uid. The second has the "shown" outcome of a lack of surrogat permissions. I illustrated this with the same command, but had the user try to use `user_admin = UserAdmin(run_as_userid="ESWIFT")` first. Unless I am misreading the target of your comment, these samples are different as they should be.

Reply:
Oh, my bad. I see the distinction. I didn't look close enough. It could be good to add a title or header for each sample, but in the interest of getting this release out, I think this is good enough.